In [1]:
import pandas as pd

In [3]:
df = pd.read_csv('Housing.csv')

In [4]:
df

price  area  bedrooms  bathrooms  stories mainroad guestroom basement  \
0    13300000  7420         4          2        3      yes        no       no   
1    12250000  8960         4          4        4      yes        no       no   
2    12250000  9960         3          2        2      yes        no      yes   
3    12215000  7500         4          2        2      yes        no      yes   
4    11410000  7420         4          1        2      yes       yes      yes   
..        ...   ...       ...        ...      ...      ...       ...      ...   
540   1820000  3000         2          1        1      yes        no      yes   
541   1767150  2400         3          1        1       no        no       no   
542   1750000  3620         2          1        1      yes        no       no   
543   1750000  2910         3          1        1       no        no       no   
544   1750000  3850         3          1        2      yes        no       no   

    hotwaterheating airconditioning  parking prefarea furnishingstatus  
0                no             yes        2      yes        furnished  
1                no             yes        3       no        furnished  
2                no              no        2      yes   semi-furnished  
3                no             yes        3      yes        furnished  
4                no             yes        2       no        furnished  
..              ...             ...      ...      ...              ...  
540              no              no        2       no      unfurnished  
541              no              no        0       no   semi-furnished  
542              no              no        0       no      unfurnished  
543              no              no        0       no        furnished  
544              no              no        0       no      unfurnished  

[545 rows x 13 columns]

In [6]:
df.columns

Index(['price', 'area', 'bedrooms', 'bathrooms', 'stories', 'mainroad',
       'guestroom', 'basement', 'hotwaterheating', 'airconditioning',
       'parking', 'prefarea', 'furnishingstatus'],
      dtype='object')

In [7]:
df.head()

price  area  bedrooms  bathrooms  stories mainroad guestroom basement  \
0  13300000  7420         4          2        3      yes        no       no   
1  12250000  8960         4          4        4      yes        no       no   
2  12250000  9960         3          2        2      yes        no      yes   
3  12215000  7500         4          2        2      yes        no      yes   
4  11410000  7420         4          1        2      yes       yes      yes   

  hotwaterheating airconditioning  parking prefarea furnishingstatus  
0              no             yes        2      yes        furnished  
1              no             yes        3       no        furnished  
2              no              no        2      yes   semi-furnished  
3              no             yes        3      yes        furnished  
4              no             yes        2       no        furnished

In [8]:
df.tail()

price  area  bedrooms  bathrooms  stories mainroad guestroom basement  \
540  1820000  3000         2          1        1      yes        no      yes   
541  1767150  2400         3          1        1       no        no       no   
542  1750000  3620         2          1        1      yes        no       no   
543  1750000  2910         3          1        1       no        no       no   
544  1750000  3850         3          1        2      yes        no       no   

    hotwaterheating airconditioning  parking prefarea furnishingstatus  
540              no              no        2       no      unfurnished  
541              no              no        0       no   semi-furnished  
542              no              no        0       no      unfurnished  
543              no              no        0       no        furnished  
544              no              no        0       no      unfurnished

In [9]:
df.head(7)

price  area  bedrooms  bathrooms  stories mainroad guestroom basement  \
0  13300000  7420         4          2        3      yes        no       no   
1  12250000  8960         4          4        4      yes        no       no   
2  12250000  9960         3          2        2      yes        no      yes   
3  12215000  7500         4          2        2      yes        no      yes   
4  11410000  7420         4          1        2      yes       yes      yes   
5  10850000  7500         3          3        1      yes        no      yes   
6  10150000  8580         4          3        4      yes        no       no   

  hotwaterheating airconditioning  parking prefarea furnishingstatus  
0              no             yes        2      yes        furnished  
1              no             yes        3       no        furnished  
2              no              no        2      yes   semi-furnished  
3              no             yes        3      yes        furnished  
4              no             yes        2       no        furnished  
5              no             yes        2      yes   semi-furnished  
6              no             yes        2      yes   semi-furnished

#EDA (exploratory data analysis)
Details that are in data set
total count is 545
average price of house is ......
maximum price 
minimum price
display price area and furnishingstatus
check wheather there is a missing value or not
if missing value handle appropriately
compare and analysis the relation between price and area
compare and analysis the relation between price and mainroad
compare and analysis the relation between price and bedrooms
one dataframe should contain price only
one dataframe should contain data other then price


In [11]:
df.value_counts().sum()

545

In [15]:
df['price'].mean()

4766729.247706422

In [17]:
round(df['price'].mean(),2)

4766729.25

In [18]:
df['price'].max()

13300000

In [19]:
df['price'].min()

1750000

In [22]:
df[['price','area','furnishingstatus']]

price  area furnishingstatus
0    13300000  7420        furnished
1    12250000  8960        furnished
2    12250000  9960   semi-furnished
3    12215000  7500        furnished
4    11410000  7420        furnished
..        ...   ...              ...
540   1820000  3000      unfurnished
541   1767150  2400   semi-furnished
542   1750000  3620      unfurnished
543   1750000  2910        furnished
544   1750000  3850      unfurnished

[545 rows x 3 columns]

In [23]:
df[['price','area','furnishingstatus','parking', 'mainroad']]

price  area furnishingstatus  parking mainroad
0    13300000  7420        furnished        2      yes
1    12250000  8960        furnished        3      yes
2    12250000  9960   semi-furnished        2      yes
3    12215000  7500        furnished        3      yes
4    11410000  7420        furnished        2      yes
..        ...   ...              ...      ...      ...
540   1820000  3000      unfurnished        2      yes
541   1767150  2400   semi-furnished        0       no
542   1750000  3620      unfurnished        0      yes
543   1750000  2910        furnished        0       no
544   1750000  3850      unfurnished        0      yes

[545 rows x 5 columns]

In [25]:
df.isnull().sum()

price               0
area                0
bedrooms            0
bathrooms           0
stories             0
mainroad            0
guestroom           0
basement            0
hotwaterheating     0
airconditioning     0
parking             0
prefarea            0
furnishingstatus    0
dtype: int64

In [27]:
df['stories'].unique()

array([3, 4, 2, 1], dtype=int64)

In [28]:
pd.crosstab(df['price'],df['stories'])

stories   1  2  3  4
price               
1750000   2  1  0  0
1767150   1  0  0  0
1820000   1  0  0  0
1855000   1  0  0  0
1890000   1  1  0  0
...      .. .. .. ..
10850000  1  0  0  0
11410000  0  1  0  0
12215000  0  1  0  0
12250000  0  1  0  1
13300000  0  0  1  0

[219 rows x 4 columns]

In [29]:
pd.crosstab(df['price'],df['mainroad'])

mainroad  no  yes
price            
1750000    1    2
1767150    1    0
1820000    0    1
1855000    1    0
1890000    0    2
...       ..  ...
10850000   0    1
11410000   0    1
12215000   0    1
12250000   0    2
13300000   0    1

[219 rows x 2 columns]

In [31]:
pd.crosstab(df['price'],df['bedrooms'])

bedrooms  1  2  3  4  5  6
price                     
1750000   0  1  2  0  0  0
1767150   0  0  1  0  0  0
1820000   0  1  0  0  0  0
1855000   0  1  0  0  0  0
1890000   0  1  1  0  0  0
...      .. .. .. .. .. ..
10850000  0  0  1  0  0  0
11410000  0  0  0  1  0  0
12215000  0  0  0  1  0  0
12250000  0  0  1  1  0  0
13300000  0  0  0  1  0  0

[219 rows x 6 columns]

In [32]:
df_n1 = df['price']

In [33]:
df_n1

0      13300000
1      12250000
2      12250000
3      12215000
4      11410000
         ...   
540     1820000
541     1767150
542     1750000
543     1750000
544     1750000
Name: price, Length: 545, dtype: int64

In [37]:
df_n2 = df.drop(['price'], axis=1)

In [38]:
df_n2

area  bedrooms  bathrooms  stories mainroad guestroom basement  \
0    7420         4          2        3      yes        no       no   
1    8960         4          4        4      yes        no       no   
2    9960         3          2        2      yes        no      yes   
3    7500         4          2        2      yes        no      yes   
4    7420         4          1        2      yes       yes      yes   
..    ...       ...        ...      ...      ...       ...      ...   
540  3000         2          1        1      yes        no      yes   
541  2400         3          1        1       no        no       no   
542  3620         2          1        1      yes        no       no   
543  2910         3          1        1       no        no       no   
544  3850         3          1        2      yes        no       no   

    hotwaterheating airconditioning  parking prefarea furnishingstatus  
0                no             yes        2      yes        furnished  
1                no             yes        3       no        furnished  
2                no              no        2      yes   semi-furnished  
3                no             yes        3      yes        furnished  
4                no             yes        2       no        furnished  
..              ...             ...      ...      ...              ...  
540              no              no        2       no      unfurnished  
541              no              no        0       no   semi-furnished  
542              no              no        0       no      unfurnished  
543              no              no        0       no        furnished  
544              no              no        0       no      unfurnished  

[545 rows x 12 columns]

In [42]:
df_n3 = df.drop(['price','airconditioning','prefarea'],axis=1)

In [43]:
df_n3

area  bedrooms  bathrooms  stories mainroad guestroom basement  \
0    7420         4          2        3      yes        no       no   
1    8960         4          4        4      yes        no       no   
2    9960         3          2        2      yes        no      yes   
3    7500         4          2        2      yes        no      yes   
4    7420         4          1        2      yes       yes      yes   
..    ...       ...        ...      ...      ...       ...      ...   
540  3000         2          1        1      yes        no      yes   
541  2400         3          1        1       no        no       no   
542  3620         2          1        1      yes        no       no   
543  2910         3          1        1       no        no       no   
544  3850         3          1        2      yes        no       no   

    hotwaterheating  parking furnishingstatus  
0                no        2        furnished  
1                no        3        furnished  
2                no        2   semi-furnished  
3                no        3        furnished  
4                no        2        furnished  
..              ...      ...              ...  
540              no        2      unfurnished  
541              no        0   semi-furnished  
542              no        0      unfurnished  
543              no        0        furnished  
544              no        0      unfurnished  

[545 rows x 10 columns]